Create /node-red/* to host NR learning resources

Yndira-E opened the pull request:
## Description

Following up on previous iteration, I've created a learning resources section structured like our docs and handbook.

This needs more work, but I think this is good enough to publish and start adding more content while I work on improving other aspects of the page.

## Related Issue(s)

[#2056](https://github.com/FlowFuse/website/issues/2056)

## Checklist

<!-- https://flowfuse.com/handbook/development/#defining-done -->

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [x] I have considered the performance impact of these changes
 - [ ] Suitable unit/system level tests have been added and they pass
 - [ ] Documentation has been updated

ZJvandeWeg commented:
@Yndira-E I don't quite understand why you scoped it to `/node-red/learning-resources/*` and why not just `/node-red/*`. I don't think we should, nor has this been discussed in our 1:1s this way.

Yndira-E commented:
Hi @ZJvandeWeg I mentioned it [here](https://github.com/FlowFuse/website/issues/1961#issuecomment-2099096530):

> Also, I was thinking that maybe we could add a secondary button, similar to the one on the homepage, that directs users to the learning resources pages.

So the button in this section: 

<img width="596" alt="Screenshot 2024-05-23 at 09 47 11" src="https://github.com/FlowFuse/website/assets/129537638/17537709-46e8-4cac-b23a-b686c259706e">

Would take you there. I realise I wasn't as specific as I probably should have been to make it clearer what I had in mind.

It just feels weird not having an "index" for those docs, but I can add a learning resources item to the nav, and it would be just another page under /node-red/

I'll change it.

ZJvandeWeg commented:
@Yndira-E dashboard.flowfuse.com uses the same path structure it seems, with a landing page at the root `/` and the rest is just `/getting-started`. :)

Yndira-E commented:
Yeah, I realised that after I wrote my previous comment!

Yndira-E commented:
@ZJvandeWeg I've implemented the eleventy plugin https://www.11ty.dev/docs/plugins/navigation/ as you suggested, it doesn't handle the groups with accordions, so I've added a script to handle that. This allow us to keep the URL as` /node-red/*` for the learning resources.

ZJvandeWeg commented on src/blog/2024/01/import-a-file.md:
```diff
@@ -30,7 +30,7 @@ This simple flow allows the user to visualize data from a CSV in the Node-RED Da
 ### Instructions ###
 1. Install Node-RED Dashboard 2.0. Follow these [instructions](/blog/2024/03/dashboard-getting-started/) to install.  
-2. Import Flow - to import the flow into your Node-RED instance follow these [instructions](/blog/2023/03/3-quick-node-red-tips-5/#1.-copy-and-share-your-flows-using-export-and-import). 
+2. Import Flow - to import the flow into your Node-RED instance follow these [instructions](/node-red/quick-tips/node-red-tips-5//#1.-copy-and-share-your-flows-using-export-and-import). 
```
```suggestion
2. Import Flow - to import the flow into your Node-RED instance follow these [instructions](/node-red/quick-tips/node-red-tips-5/#1.-copy-and-share-your-flows-using-export-and-import). 
```

ZJvandeWeg commented on src/redirect.md:
```diff
@@ -44,10 +44,17 @@ redirects:
   - { "from": "/product/", "to": "/product/features/" }
   - { "from": "/product/why-flowfuse/", "to": "/product/features/" }
   - { "from": "/features/", "to": "/product/features/" }
-  - { "from": "/node-red/core-nodes/rbe/", "to": "/node-red/core-nodes/filter/" }
```
Don't remove please

Yndira-E commented on .eleventy.js:
```diff
@@ -97,6 +98,15 @@ module.exports = function(eleventyConfig) {
+    const coreNodes = JSON.parse(fs.readFileSync(path.resolve(__dirname, 'src/_data/coreNodes.json'), 'utf-8'));
+
+    // Transform coreNodes object into an array
+    const coreNodesArray = Object.entries(coreNodes).map(([key, nodes]) => ({ key, nodes }));
```
Done

Yndira-E commented on src/node-red/core-nodes/nav.njk:
```diff
@@ -0,0 +1,15 @@
+eleventyComputed:
+    eleventyNavigation:
+        key: "{{ category.key }}"
+        title: "{{ category.key.charAt(0).toUpperCase() + category.key.slice(1) }}"
```
You're right, I've replaced it with the filter

Yndira-E commented on .eleventy.js:
```diff
@@ -97,6 +98,15 @@ module.exports = function(eleventyConfig) {
     });
 
+    // Read the JSON file with core nodes
+    const coreNodes = JSON.parse(fs.readFileSync(path.resolve(__dirname, 'src/_data/coreNodes.json'), 'utf-8'));
```
Done

ZJvandeWeg commented on src/_includes/layouts/learning-resources-nav.njk:
```diff
@@ -0,0 +1,70 @@
+    {% macro renderNavListItem(entry) -%}
+        {% set isChildUrl = false %}
+        {% for child in entry.children %}
+            {% if child.url == page.url %}
+                {% set isChildUrl = true %}
+            {% endif %}
```
```suggestion
            {% set isChildUrl = child.url == page.url %}
```

ZJvandeWeg commented on src/_includes/layouts/learning-resources-nav.njk:
```diff
@@ -0,0 +1,70 @@
+            {% endif %}
+        {% endfor %}
+
+        <li class="{% if entry.url==page.url and not isChildUrl %}active{% endif %}{% if entry.children.length %} has-children{% endif %}">
```
Why can childURLs not be active?

Yndira-E replied:
This is related to the classes of the core-nodes categories, because we don't have an index file and the name of the category redirects to the first child. If I didn't add this condition, when the first child was active, the parent remained active as well since they have the same URL

ZJvandeWeg commented on src/_includes/layouts/learning-resources.njk:
```diff
@@ -0,0 +1,112 @@
+    </div>
+</div>
+</div>
+<script>
```
This is pure duplication from an other page. let's make partials

ZJvandeWeg commented on src/node-red/learning-resources.md:
```diff
@@ -0,0 +1,19 @@
+---
```
What's this page for? It feels like a summery of the left nav bar?

ZJvandeWeg commented on src/node-red/quick-tips/node-red-tips-7.md:
```diff
@@ -66,11 +63,4 @@ If you'd like to view this slider and the flow which makes it work on your own N
 
-We hope you found these tips useful, if you'd like to suggest some of your own tips which you think we should share in our future blog posts please [get in touch](mailto:contact@flowfuse.com). You can also read some of our previous Node-RED tips using the links below.
-
-[Node-RED Tips - Subflows, Link Nodes, and the Range Node](/blog/2023/04/3-quick-node-red-tips-6/)\
-[Node-RED Tips - Importing, Exporting, and Grouping Flows](/blog/2023/03/3-quick-node-red-tips-5/)\
-[Node-RED Tips - Smooth, Catch, and Maths](/blog/2023/03/3-quick-node-red-tips-4/)\
-[Node-RED Tips - Exec, Filter, and Debug](/blog/2023/03/3-quick-node-red-tips-3/)\
-[Node-RED Tips - Deploying, Debugging, and Delaying](/blog/2023/02/3-quick-node-red-tips-2/)\
-[Node-RED Tips - Wiring Shortcuts](/blog/2023/02/3-quick-node-red-tips-1/)
```
Why remove this?

Yndira-E commented:
Replaced by https://github.com/FlowFuse/website/pull/2113
